### ISSUE TITLE
scatter phase function x 4 PI

### ISSUE THREAD

**Issue by lucasbarbedo83** [CNES/RadiativeTransferCode-OSOAA#15]
in The OSAA manual, pg. 81:
 4.2.2.1 Radiative properties of an individual component
It says that the spatial integral of the volume phase function is normalized by 4*PI to equate to 1.
Does it mean I should multiply my external phase function by 4*PI before it is ingested in OSOAA? 
Or the OSAA model will do that internally.
A+ Lucas

**Comment by lucasbarbedo83** [comment]
The second question is about scattering. 
How is the scattering coefficient is propagated throughout the OSOAA model?
I was tracking the single scattering albedo, for example, the variable is PIZ.
Let's consider, an ocean with only seawater and phytoplankton.
In OSOAA_HYDROSOL, for example, the single albedo is PIZ=KMAT2/KMAT1 (line 1177) means 
the ratio scatter and attenuation cross-sections, which is equal b_p/c_p (particle scattering coef./particle attenuation coef.).
In OSOAA_PROFILE (line 859),  total extinction is calculated as:
  CEXT_TOT=MOL_ABS+MOL_DIF+SED_ABS+SED_DIF+YS_ABS+
     &	 	  DET_ABS+PROF_PHY_ABS(I)+PROF_PHY_DIF(I).
The parameter PROF_PHY_DIF and PROF_PHY_ABS  are calculated from an empirical relation between chlorophyll concentration and scattering coefficient and phytoplankton absorption, respectively (line 1830).
I notice that when Hyd.Mod=1 is selected, and OSOAA produces scatter and attenuation cross-sections with the same value, as we can check in the respective log file.
The question: when is the scattering coefficient incorporated on phase function as  bp x P11?
 In other words, should I ingest the phase function (P11) or volume scattering function  (P11xbp)  on the External file (HyD.Mod=2).
![image](https://user-images.githubusercontent.com/101749383/226076426-ee571665-1391-47c6-b8db-8fc9a7168903.png)
A+ Lucas

**Comment by blafranc** [comment]
Dear Lucas,

a)	The routine OSOAA_PROFILE.F calculate the mixing rates which include the scattering coefficients of particles (see tables PROF_Z_MEL_PHY for the phytoplankton, PROF_Z_MEL_SED for the mineral-like particles, PROF_Z_MEL_MOL for the water molecules).

b)	These mixing rates are adjusted to the truncation of their phase functions by the subroutine OSOAA_SOS.

c)	They are then used to solve the radiative transfer equation in the routine OSOAA_SOS_CORE:  the routine FSOURCE_SEA_ORDRE1 calculates the source function for the direct solar beam, and OSOAA_FSOURCE_SEA_ORDREIG calculates the source function for the diffuse light.
Therefore, this is where the scattering coefficient of the hydrosols is taken into account in the calculation of the resulting phase function of the hydrosol mixture.
The single scattering albedo is introduced in the source functions by these calculations.

To sum up, **you must introduce the phase function (unit in sr-1)**, but not the volume scattering function (m-1 sr-1), as the user phase function.
This is indicated in page 36 (table 6) for the use of the parameter HYD.ExtData

Indeed, **you must ensure the spatial integration of your phase function is equal to 4pi** (according to the equation in page 81 of the user manual). This requirement is not checked by OSOAA.

Best regards

**Comment by lucasbarbedo83** [comment]
All phase functions, or just the P11 component?
P12/P11 P22/P11.... Must I multiply by 4pi?

**Comment by blafranc** [comment]
All the phase functions of the Mueller matrix must be normalized in the same way than the P11 phase function.
That means, if the P11 phase function must be corrected by a factor F, the polarized Pij phase functions must be corrected by the same factor.

**Comment by lucasbarbedo83** [comment]
ok,
However, we have as an input, for example:

_EXTINCTION_COEF : 0.00003 
SCATTERING_COEF : 0.00003 
NB_LINES : 83 
ANGLE F11 -F12/F11 F22/F11 F33/F11 
180.00 0.029000 -0.000000 1.000000 -1.000000 
178.29 0.019730 0.018419 1.000000 -0.368170 
176.07 0.015810 0.014231 1.000000 -0.674257 
173.84 0.014650 0.017392 1.000000 -0.791809 
171.61 0.014170 0.016246 1.000000 -0.845448_ 
....
Must It be calculated as below: 
ANGLE 4*pi*F11   4*pi*(-F12/F11)   4*pi*(F22/F11) 4*pi*(F33/F11)


A+ Lucas

**Comment by blafranc** [comment]
Dear Lucas,

Please find some explanations in the attached file.
[Phase_function_normalization.pdf](https://github.com/CNES/RadiativeTransferCode-OSOAA/files/11091159/Phase_function_normalization.pdf)

The second file can give you a view of the geometry modeled in OSOAA.
[Geometry_Illustration.pdf](https://github.com/CNES/RadiativeTransferCode-OSOAA/files/11091228/Geometry_Illustration.pdf)

**Comment by lucasbarbedo83** [comment]
![image](https://user-images.githubusercontent.com/101749383/228308272-b6c9f138-2f9e-463a-97cd-8ad5ae1b3516.png)
The solid angle is not correct.

**Comment by lucasbarbedo83** [comment]
I have tried many options, I try to read the phase function OSOAA print out. 
It is not making sense to me.

The table below contains a real in-situ Mueller Matrix at 517 nm
We measured on the Mississippi Coast. 
It is not multiplied by any convection factor, for example, 4pi.
In case you, integrate the P11 (i.e. volume scattering function)
![image](https://user-images.githubusercontent.com/101749383/228314690-1d7b15d3-45e5-4e9e-ab90-f55b70219d74.png)

the value will be the scattering coefficient below 3.59561 m-1.
We have an above-water reflectance Rrs(0+,517) = 0.008 sr-1.

Please, try to run OSOAA v1.6  using these input values. Chlorophyll concentration: 0..719 mg m-3

EXTINCTION_COEF : 4.36652 
SCATTERING_COEF : 3.59561 
NB_LINES : 198 
ANGLE F11 -F12/F11 F22/F11 F33/F11 
0.01 23461.312009 0.000001 1.000000 1.000000 
0.02 19426.282496 0.000003 1.000000 1.000000 
0.03 14653.282751 0.000006 1.000000 1.000000 
0.04 10633.534184 0.000007 1.000000 1.000000 
0.05 7823.309229 0.000005 1.000000 1.000000 
0.06 5938.558929 0.000004 1.000000 1.000000 
0.07 4583.842402 0.000003 1.000000 1.000000 
0.08 3560.503590 0.000004 1.000000 0.999999 
0.09 2801.937205 0.000005 1.000000 0.999999 
0.10 2250.836725 0.000007 0.999999 0.999999 
0.20 618.625151 0.000006 0.999998 0.999997 
0.30 331.220118 0.000005 0.999996 0.999996 
0.40 221.511391 0.000008 0.999995 0.999994 
0.50 163.192369 0.000010 0.999993 0.999992 
0.60 126.742346 0.000017 0.999991 0.999990 
0.70 101.990223 0.000017 0.999989 0.999987 
0.80 84.288742 0.000027 0.999986 0.999985 
0.90 71.101804 0.000039 0.999984 0.999983 
1.00 60.979878 0.000045 0.999982 0.999980 
2.00 22.137546 0.000293 0.999956 0.999954 
3.00 11.832900 0.000857 0.999923 0.999921 
4.00 7.394736 0.001730 0.999883 0.999878 
5.00 4.947412 0.002987 0.999829 0.999820 
6.00 3.491603 0.004551 0.999759 0.999743 
7.00 2.512119 0.006375 0.999665 0.999636 
8.00 1.813936 0.008422 0.999530 0.999481 
9.00 1.316484 0.010651 0.999340 0.999264 
10.00 0.959484 0.013137 0.999071 0.998959 
11.00 0.702972 0.015800 0.998689 0.998526 
12.00 0.518083 0.018702 0.998138 0.997908 
13.00 0.386667 0.021858 0.997372 0.997057 
14.00 0.294097 0.025397 0.996388 0.995957 
15.00 0.229162 0.029366 0.995148 0.994579 
16.00 0.182695 0.033519 0.993610 0.992862 
17.00 0.149021 0.038095 0.991776 0.990822 
18.00 0.124036 0.042955 0.989663 0.988466 
19.00 0.104982 0.048079 0.987342 0.985871 
20.00 0.090065 0.053095 0.984838 0.983058 
21.00 0.077880 0.058447 0.982191 0.980034 
22.00 0.067913 0.063969 0.979586 0.977034 
23.00 0.059389 0.069924 0.976998 0.973938 
24.00 0.052449 0.075797 0.974623 0.971042 
25.00 0.046539 0.081934 0.972406 0.968234 
26.00 0.041538 0.089135 0.970414 0.965483 
27.00 0.037421 0.096033 0.968855 0.963149 
28.00 0.033908 0.103164 0.967628 0.960996 
29.00 0.030952 0.111174 0.966781 0.959141 
30.00 0.028363 0.118970 0.966293 0.957508 
31.00 0.026041 0.127505 0.966046 0.955958 
32.00 0.023971 0.136871 0.966107 0.954586 
33.00 0.022115 0.146763 0.966461 0.953354 
34.00 0.020436 0.157024 0.967068 0.952296 
35.00 0.018852 0.168070 0.967764 0.950950 
36.00 0.017479 0.178438 0.968749 0.949663 
37.00 0.016199 0.190521 0.969809 0.948371 
38.00 0.015035 0.201845 0.970995 0.946884 
39.00 0.013938 0.214833 0.972214 0.944989 
40.00 0.012970 0.227597 0.973597 0.943254 
41.00 0.012079 0.241242 0.975005 0.940982 
42.00 0.011300 0.253515 0.976508 0.938475 
43.00 0.010574 0.268464 0.977994 0.936100 
44.00 0.009882 0.282993 0.979389 0.932983 
45.00 0.009250 0.297977 0.980713 0.929409 
46.00 0.008682 0.313480 0.981965 0.925581 
47.00 0.008142 0.328315 0.983091 0.920743 
48.00 0.007660 0.343635 0.984180 0.915718 
49.00 0.007207 0.359417 0.985149 0.909865 
50.00 0.006819 0.373709 0.986134 0.904584 
51.00 0.006417 0.392535 0.986924 0.897715 
52.00 0.006049 0.408633 0.987627 0.890170 
53.00 0.005730 0.423578 0.988330 0.882233 
54.00 0.005401 0.443686 0.988886 0.873880 
55.00 0.005113 0.458484 0.989422 0.864217 
56.00 0.004838 0.478683 0.989902 0.854924 
57.00 0.004592 0.494808 0.990356 0.844756 
58.00 0.004361 0.511485 0.990763 0.834072 
59.00 0.004137 0.529130 0.991109 0.822327 
60.00 0.003928 0.547089 0.991445 0.809970 
61.00 0.003732 0.565034 0.991754 0.797102 
62.00 0.003544 0.582406 0.992031 0.783230 
63.00 0.003369 0.600796 0.992281 0.768726 
64.00 0.003203 0.618698 0.992520 0.753538 
65.00 0.003048 0.636587 0.992725 0.737844 
66.00 0.002900 0.653580 0.992893 0.720424 
67.00 0.002765 0.671534 0.993063 0.703127 
68.00 0.002638 0.688842 0.993209 0.684879 
69.00 0.002516 0.704643 0.993333 0.665305 
70.00 0.002403 0.721187 0.993431 0.645969 
71.00 0.002297 0.736688 0.993531 0.625442 
72.00 0.002197 0.752158 0.993607 0.604305 
73.00 0.002101 0.766810 0.993668 0.582602 
74.00 0.002015 0.780848 0.993721 0.559793 
75.00 0.001934 0.793153 0.993769 0.537539 
76.00 0.001854 0.804357 0.993803 0.513603 
77.00 0.001783 0.814266 0.993828 0.490292 
78.00 0.001717 0.824655 0.993861 0.466911 
79.00 0.001658 0.829622 0.993894 0.445370 
80.00 0.001587 0.844644 0.993869 0.418135 
81.00 0.001506 0.871369 0.993775 0.383737 
82.00 0.001428 0.897123 0.993671 0.345867 
83.00 0.001359 0.917495 0.993575 0.307777 
84.00 0.001297 0.934040 0.993499 0.270355 
85.00 0.001244 0.947191 0.993438 0.233977 
86.00 0.001195 0.957752 0.993390 0.197884 
87.00 0.001150 0.965283 0.993340 0.162208 
88.00 0.001110 0.972379 0.993311 0.126282 
89.00 0.001070 0.977074 0.993269 0.090628 
90.00 0.001033 0.980177 0.993236 0.054807 
91.00 0.001000 0.982647 0.993216 0.019079 
92.00 0.000969 0.982885 0.993195 -0.016319 
93.00 0.000941 0.982187 0.993190 -0.051645 
94.00 0.000913 0.979976 0.993180 -0.086988 
95.00 0.000887 0.976362 0.993170 -0.121793 
96.00 0.000863 0.972080 0.993169 -0.156544 
97.00 0.000841 0.966219 0.993177 -0.191223 
98.00 0.000819 0.958998 0.993177 -0.224786 
99.00 0.000800 0.951029 0.993189 -0.258044 
100.00 0.000782 0.941618 0.993207 -0.291056 
101.00 0.000764 0.931154 0.993213 -0.322840 
102.00 0.000747 0.919905 0.993228 -0.354456 
103.00 0.000732 0.907662 0.993246 -0.385100 
104.00 0.000718 0.894521 0.993264 -0.415333 
105.00 0.000705 0.880533 0.993284 -0.444440 
106.00 0.000693 0.865891 0.993296 -0.473057 
107.00 0.000681 0.850404 0.993322 -0.500509 
108.00 0.000670 0.834241 0.993325 -0.527068 
109.00 0.000661 0.817636 0.993353 -0.552901 
110.00 0.000652 0.800333 0.993373 -0.577934 
111.00 0.000643 0.782656 0.993375 -0.601882 
112.00 0.000635 0.764421 0.993393 -0.624849 
113.00 0.000627 0.745907 0.993383 -0.646853 
114.00 0.000622 0.727284 0.993411 -0.668021 
115.00 0.000616 0.708120 0.993417 -0.688392 
116.00 0.000609 0.688821 0.993402 -0.707758 
117.00 0.000605 0.669543 0.993428 -0.726133 
118.00 0.000601 0.650079 0.993424 -0.743542 
119.00 0.000596 0.630606 0.993435 -0.760115 
120.00 0.000593 0.610940 0.993435 -0.776177 
121.00 0.000589 0.591827 0.993422 -0.790833 
122.00 0.000586 0.572740 0.993438 -0.804662 
123.00 0.000581 0.553604 0.993405 -0.818011 
124.00 0.000579 0.534736 0.993434 -0.830473 
125.00 0.000576 0.516118 0.993434 -0.842188 
126.00 0.000573 0.497778 0.993439 -0.853057 
127.00 0.000571 0.479462 0.993460 -0.863457 
128.00 0.000568 0.461539 0.993436 -0.873097 
129.00 0.000566 0.443926 0.993456 -0.882201 
130.00 0.000564 0.426461 0.993455 -0.890644 
131.00 0.000563 0.409428 0.993478 -0.898544 
132.00 0.000561 0.392792 0.993452 -0.905870 
133.00 0.000560 0.376447 0.993481 -0.912733 
134.00 0.000558 0.360396 0.993453 -0.919121 
135.00 0.000557 0.344782 0.993457 -0.925108 
136.00 0.000556 0.329628 0.993457 -0.930418 
137.00 0.000555 0.314767 0.993441 -0.935511 
138.00 0.000557 0.300417 0.993454 -0.940229 
139.00 0.000554 0.286291 0.993415 -0.944424 
140.00 0.000554 0.272729 0.993396 -0.948243 
141.00 0.000553 0.259536 0.993351 -0.951893 
142.00 0.000553 0.246427 0.993326 -0.955222 
143.00 0.000555 0.234050 0.993275 -0.958165 
144.00 0.000554 0.222441 0.993218 -0.960693 
145.00 0.000553 0.210673 0.993129 -0.963127 
146.00 0.000554 0.199752 0.993048 -0.965312 
147.00 0.000553 0.188959 0.992938 -0.967086 
148.00 0.000554 0.178534 0.992814 -0.968734 
149.00 0.000556 0.169139 0.992642 -0.970124 
150.00 0.000557 0.160229 0.992412 -0.970858 
151.00 0.000558 0.152429 0.992137 -0.971086 
152.00 0.000559 0.143570 0.991910 -0.972307 
153.00 0.000559 0.134868 0.991707 -0.973408 
154.00 0.000559 0.126415 0.991468 -0.974395 
155.00 0.000560 0.118198 0.991287 -0.975465 
156.00 0.000562 0.110551 0.991031 -0.976250 
157.00 0.000562 0.103200 0.990695 -0.977103 
158.00 0.000562 0.096148 0.990376 -0.977811 
159.00 0.000563 0.088946 0.989958 -0.978640 
160.00 0.000563 0.082205 0.989578 -0.979351 
161.00 0.000564 0.075634 0.989028 -0.980020 
162.00 0.000565 0.069334 0.988617 -0.980971 
163.00 0.000567 0.063128 0.987986 -0.981625 
164.00 0.000568 0.057449 0.987281 -0.982198 
165.00 0.000570 0.051456 0.986580 -0.982728 
166.00 0.000570 0.045920 0.986068 -0.983073 
167.00 0.000572 0.040237 0.985889 -0.983488 
168.00 0.000573 0.034831 0.985598 -0.983837 
169.00 0.000574 0.030496 0.985246 -0.983478 
170.00 0.000575 0.025925 0.984856 -0.983262 
171.00 0.000576 0.022215 0.984293 -0.982849 
172.00 0.000577 0.018032 0.983610 -0.982255 
173.00 0.000578 0.014832 0.983026 -0.981918 
174.00 0.000580 0.011769 0.982263 -0.981244 
175.00 0.000582 0.010452 0.981324 -0.980272 
176.00 0.000585 0.008510 0.980500 -0.978985 
177.00 0.000586 0.006599 0.979382 -0.977182 
178.00 0.000586 0.004895 0.978440 -0.975687 
179.00 0.000586 0.004470 0.978497 -0.972175 
180.00 0.000684 -0.000000 0.981954 -0.981954

**Comment by lucasbarbedo83** [comment]
OSOAA_MAIN.exe 
-OSOAA.ResRoot  {LOCAL}/OSOAA_RESULTS/0.517 -OSOAA.Log Main.Log -OSOAA.Wa 0.517 -ANG.Thetas 44.720 -AP.Pressure 1013.00 -AP.HR 5.0 -AP.HA 2.00 -AER.Waref 0.550 -AER.AOTref 0.10 -AER.DirMie  {LOCAL}/DATABASE/MIE_AER -AER.Model 2 -SEA.Depth 100.00 -AER.SF.Model 3 -AER.SF.RH 1 -HYD.DirMie {LOCAL}/DATABASE/MIE_HYD -HYD.Model 2 -HYD.ExtData {LOCAL}/HYD_ExtData/HYD_ExtData_0.517.txt -HYD.Log HYDlog_MuellerMatrix.txt -HYD.MieLog HYDMieLog.txt -SEA.Dir {LOCAL}//DATABASE/SURF_MATR -SEA.Ind 1.34 -SEA.Wind 4.00 -SEA.SurfAlb 0.0 -SEA.BotType 2 -OSOAA.View.Phi 45 -OSOAA.View.Level 3 -OSOAA.ResFile.vsVZA RESLUM_vsVZA_wa0.517ViewPhi45.txt -OSOAA.ResFile.Adv.Up RESLUM_Advanced_UP.txt -OSOAA.ResFile.Adv.Down RESLUM_Advanced_DOWN_wa0.517.txt -SED.Csed 0.00 -YS.Abs440 0 -DET.Abs440 0.227 -PHYTO.ProfilType 1 -PHYTO.Chl 0.719 -YS.Swa -0.008 -DET.Swa -0.008

**Comment by yichen-li2016** [comment]
Hi Lucas,
@lucasbarbedo83

The value of the SCATTERING_COEF of the second line of the ExtData you gave above is equal to the scattering coefficient in **m-1**. However, according to the OSOAA user manual, the SCATTERING_COEF should be **the scattering cross-sectional area** in um-2.

In addition to that, I see that the scattering phase function you used is measured from the Mississippi Coast, which should include inorganic particle scattering, but only the CHL concentration is set in your processing command line, how is your scattering of inorganic particulate matter considered?

Best regards,

Qiang
